test_expect_failure/scalacopts_from_toolchain/BUILD

```diff
@@ -17,5 +17,5 @@ toolchain(
 scala_library(
     name = "failing_build",
     srcs = ["ClassWithUnused.scala"],
-    scalacopts = ["-Xfatal-warnings"],
+    scalacopts = ["-Yfatal-warnings"],
```

Comment:
why did you do this change? Pretty sure it's by intention

Reply:
The `test_toolchain.sh` test did not pass unless this change happened, it certainly was the one that confused me the most

Reply:
Ok, maybe share what is the output when it fai? This is xfatal because you want to know that the scalacopts are taken into consideration. The way we know is by asking to fail on warnings and inserting a warning there.

Reply:
Can you share the output of the failure before? Moving from -X to -Y means this might break, right? Also I mainly want to verify this change didn't change the semantics of the test

Reply:
```
INFO: Found 1 target...
[2 / 21] [Prepa] BazelWorkspaceStatusAction stable-status.txt
[199 / 213] checking cached actions
[212 / 213] scala //test_expect_failure/scalacopts_from_toolchain:failing_build; 0s worker
INFO: From scala //test_expect_failure/scalacopts_from_toolchain:failing_build:
test_expect_failure/scalacopts_from_toolchain/ClassWithUnused.scala:5: warning: local val unusedValue in method talk is never used
    val unusedValue = "I am not used :-("
        ^
Target //test_expect_failure/scalacopts_from_toolchain:failing_build up-to-date:
  bazel-bin/test_expect_failure/scalacopts_from_toolchain/failing_build.jar
INFO: Elapsed time: 18.015s, Critical Path: 3.84s
INFO: 1 process: 1 worker.
INFO: Build completed successfully, 3 total actions
INFO: Build completed successfully, 3 total actions
 "bazel build --extra_toolchains=//test_expect_failure/scalacopts_from_toolchain:failing_scala_toolchain //test_expect_failure/scalacopts_from_toolchain:failing_build" should have failed but passed.
```

Reply:
My solution is definitely wrong, since I just checked it passed due to `-Yfatal-warnings` being a bad option. However, I don't know how this fails. I'm going to check if master shares this issue or if I broke it.

Reply:
good luck

Reply:
Sorry for taking so long, this was a bit complicated. Two problems were found regarding the reporters. First of all, the severity is defined a bit strangely,  as you can see [here](https://github.com/scala/scala/blob/8684ae833dcfeac6107343fcca5501301e509eef/src/compiler/scala/tools/nsc/reporters/Reporter.scala#L48). Error severity is defined differently from the others, which does lead to some problems in the end of things, since the ConsoleReporter defines [this function](https://github.com/scala/scala/blob/8684ae833dcfeac6107343fcca5501301e509eef/src/compiler/scala/tools/nsc/reporters/ConsoleReporter.scala#L48) which is called by its parent when it received a new diagnostic, that is when [this function](https://github.com/scala/scala/blob/8684ae833dcfeac6107343fcca5501301e509eef/src/compiler/scala/tools/nsc/reporters/AbstractReporter.scala#L35) is called by its parent. With composition, this lead to different instances of `ERROR` that were not comparable. Before finding error, we were calling directly the functions `warning`, `error` and `info`. This later turned out not to be possible since the info method of abstract reporter had some logic that pertained to several compiler flags, such as the the one used above. 
So we knew we had to call info directly for the `ConsoleReporter` to be able to properly detect which kind of error it was. However, due to the lazy val definition of the ERROR severity, it had to come from the same instance. One could just instantiate the error directly but then we risk the possibility of losing any information that the compiler might put into ERROR by overriding that definition (this proved to be a problem for warnings for example, making this test fail in that circumstance). The only thing that was left for us was to extend `ConsoleReporter` directly, and repeat some of the code existing in the `AbstractReporter`, since you can't call the info method of `AbstractReporter` when you're overriding it as well.

Reply:
I hope it's clear enough, since this one is a bit tricky